### PR TITLE
[13.0][IMP] website_sale_hide_price: hide price at product level

### PR DESCRIPTION
**Pull request by chienandalu**
- [x] Now we can hide the price only for the products we want to.
- [x] Adding this new flag to the comparison compatibility module
- [x] Improved readme
- [x] Fixed product price visible in recently viewed products

cc @Tecnativa TT36129

Ready to review: @CarlosRoca13 @victoralmau

### CODE REVIEW

**Comment by chienandalu** [comment]
It can be reviewed/tested

File: website_sale_hide_price/static/src/xml/website_sale_templates.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <t t-jquery="div[t-if='widget.displayPrice']" t-operation="attributes">
+        <attribute
+                name="t-if"
+                value="widget.displayPrice &amp; !product.website_hide_price"
+            />
+    </t>
+</t>
+<t t-extend="website_sale.productsRecentlyViewed">
+    <t t-jquery="div[t-raw='product.price']" t-operation="attributes">
+        <attribute name="t-if" value="!product.website_hide_price" />
```
Comment (CarlosRoca13) [inline website_sale_hide_price/static/src/xml/website_sale_templates.xml:15]: You are just considering when the product has the check.

In next image you can see it:
imagen

The partner of Mitchell Admin has the prices hidden and the product Customizable Desk too. When I go to see the recently products, I can see the prices of other products. (You can check it on runboat)

**Review by CarlosRoca13 — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@CarlosRoca13 your re-review is pending

**Comment by CarlosRoca13** [comment]
But no changes were added since my last review

**Comment by pedrobaeza** [comment]
Ah, OK. @chienandalu then it's your call.

File: website_sale_hide_price/static/src/xml/website_sale_templates.xml (hunk 1 above)
Comment (chienandalu) [reply website_sale_hide_price/static/src/xml/website_sale_templates.xml:15]: Yes, it's a limitation. In that template we don't have the website object. We'd need to inject the global variable somehow

Comment (chienandalu) [reply website_sale_hide_price/static/src/xml/website_sale_templates.xml:15]: Ok, I found a way to fix it :) 

You can review now

**Comment by pedrobaeza** [comment]
@CarlosRoca13 @willytal please review

**Review by CarlosRoca13 — APPROVED** [review]
Thanks! :smile:

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please fw-port it to v14 and v15

**Comment by CarlosRoca13** [comment]
In v15 it is solved at migration, so it shouldn't be necessary :smile:

**Comment by sergioeix** [comment]
He probado y testeando la nueva versión de este modulo es genial poder trabajar a nivel del producto pero he descubierto una forma de poder ver los precios y en carrito, (pones el public_user en hide y accedes a la tienda sin registrarte)  aunque no es una función que yo gaste si tienes lista de deseos es posible añadir un producto en esta lista y entonces puedes ver el carrito con el precio.
Un saludo.

**Comment by chienandalu** [comment]
Probáblemente haría falta un módulo de compatibilidad con el de wishlist.

**Pull request merged**